staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -252,6 +252,47 @@ type PodFailurePolicy struct {
 	Rules []PodFailurePolicyRule `json:"rules" protobuf:"bytes,1,opt,name=rules"`
 }
 
+// SuccessPolicy describes when a Job can be declared as succeeded based on the success of some indexes.
+type SuccessPolicy struct {
+	// Criteria represents the list of alternative criteria for the declaring the Jobs
+	// as successful before its completion. Once any of the criteria are met,
+	// the "SucceededCriteriaMet" condition is added, and the lingering pods are removed.
+	// The terminal state for such a Job has the "Completed" condition.
+	// Additionally, these criteria are evaluated in order; Once the Job meets one of the criteria,
+	// other criteria are ignored.
+	// +listType=atomic
+	Criteria []SuccessPolicyCriteria `json:"criteria" protobuf:"bytes,1,opt,name=criteria"`
+}
+
+// SuccessPolicyCriteria describes criterion for declaring a Job as succeeded.
+// Each criterion must have at least one of the "succeededIndexes" or "succeededCount" specified.
+type SuccessPolicyCriteria struct {
+	// SucceededIndexes specifies the set of indexes
+	// which need to be contained in the actual set of the succeeded indexes for the Job.
+	// The list of indexes must be within 0 to ".spec.completions-1" and
+	// must not contain duplicates. At least one element is required.
+	// The indexes are represented as decimal integers separated by commands.
+	// The number are listed in represented by the first and last element of the series,
+	// separated by a hyphen.
+	// For example, if the completed indexes are 1, 3, 4, 5 and 7, they are
+	// represented as "1,3-5,7".
+	// When this field is null, this field doesn't default to any value
+	// and is never evaluated at any time.
+	//
+	// +optional
+	SucceededIndexes *string `json:"succeededIndexes,omitempty" protobuf:"bytes,1,opt,name=succeededIndexes"`
+
+	// SucceededCount specifies the minimal required size of the actual set of the succeeded indexes
+	// for the Job. When SucceededCount is used along with SucceededIndexes, the check is
+	// constrained only to the set of indexes specified by SucceededIndexes.
```

Comment:
This reads strangely.  Can I get an example?

Say
SucceededIndex: "1-4"
SucceededCount: 3
Completions:1, 3, 5

Success or not yet?

Reply:
That's not yet.
The examples are the following:

```
# Success
SucceededIndex: "1-4"
SucceededCount: 3
Completions:1, 3, 4
```

```
# Success
SucceededCount: 3
Completions:1, 3, 5
```

Reply:
Done.